src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -1,29 +1,85 @@
  */
 public class DeleteCommandParser implements Parser<DeleteCommand> {
 
+
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
-        try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
-        } catch (ParseException pe) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_CLIENTID, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
+                    PREFIX_ADDRESS, PREFIX_RISKAPPETITE, PREFIX_DISPOSABLEINCOME, PREFIX_CURRENTPLAN, PREFIX_LASTMET,
+                    PREFIX_TAG);
+
+        if (((!arePrefixesPresent(argMultimap, PREFIX_CLIENTID)) && (!arePrefixesPresent(argMultimap, PREFIX_EMAIL)))
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
+        }
+
+        //Throws error if other fields are inputted
+        if (arePrefixesPresent(argMultimap, PREFIX_NAME) || arePrefixesPresent(argMultimap, PREFIX_PHONE)
+            || arePrefixesPresent(argMultimap, PREFIX_RISKAPPETITE)
+            || arePrefixesPresent(argMultimap, PREFIX_ADDRESS)
+            || arePrefixesPresent(argMultimap, PREFIX_DISPOSABLEINCOME)
+            || arePrefixesPresent(argMultimap, PREFIX_CURRENTPLAN)
+            || arePrefixesPresent(argMultimap, PREFIX_LASTMET)) {
+            throw new ParseException(String.format(Messages.MESSAGE_TOO_MANY_FIELDS, DeleteCommand.MESSAGE_USAGE));
+        }
+
+        ArrayList<Predicate> predicatesToDelete = new ArrayList<>();
+        String clientIdString = argMultimap.getValue(PREFIX_CLIENTID).orElse("");
+        if (!clientIdString.isEmpty()) {
+            ClientId clientId = ParserUtil.parseClientId(clientIdString);
+            predicatesToDelete.add(new PersonHasId(clientId));
         }
+
+        String emailString = argMultimap.getValue(PREFIX_EMAIL).orElse("");
+        if (!emailString.isEmpty()) {
+            Email email = ParserUtil.parseEmail(emailString);
+            predicatesToDelete.add(new PersonHasEmail(email));
+        }
+
+        return new DeleteCommand(predicatesToDelete);
+
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```

Comment:
Perhaps change to anyMatch which would be more useful for both the above usage. You can make use of the Varargs above to check for multiple prefix at once
```suggestion
        return Stream.of(prefixes).anyMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
```

Reply:
Thanks for the suggession! I implemented it as a new function